src/daft-local-execution/src/intermediate_ops/intermediate_op.rs

```diff
@@ -3,9 +3,132 @@ use std::sync::Arc;
 use common_error::DaftResult;
 use daft_micropartition::MicroPartition;
 
+use crate::{
+    create_channel, create_single_channel, MultiReceiver, MultiSender, SingleReceiver,
+    SingleSender, NUM_CPUS,
+};
+
 pub trait IntermediateOperator: dyn_clone::DynClone + Send + Sync {
     fn execute(&self, input: &Arc<MicroPartition>) -> DaftResult<Arc<MicroPartition>>;
     fn name(&self) -> &'static str;
 }
 
 dyn_clone::clone_trait_object!(IntermediateOperator);
+
+/// The number of rows that will trigger an intermediate operator to output its data.
+pub const OUTPUT_THRESHOLD: usize = 1000;
+
+/// State of an operator task, used to buffer data and output it when a threshold is reached.
+pub struct OperatorTaskState {
```

Comment:
No action needed but we should refactor this to be a trait that is serde-able so diff operators can have diff impls.